cmd/ethkey/main.go

```diff
@@ -0,0 +1,322 @@
+package main
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/cmd/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/pborman/uuid"
+	"gopkg.in/urfave/cli.v1"
+)
+
+const (
+	version   = "0.1.0"
+	copyright = "Copyright 2017 The go-ethereum Authors"
+
+	defaultKeyfilePath = "keyfile.json"
+)
+
+var (
+	app *cli.App // the main app instance
+)
+
+var ( // Commonly used command line flags.
+	passphraseFlag = cli.StringFlag{
+		Name:  "passphrase",
+		Usage: "The passphrase for the keyfile.",
+	}
+
+	passphrasefileFlag = cli.StringFlag{
+		Name:  "passwordfile",
+		Usage: "The file that contains the passphrase for the keyfile.",
+	}
+)
+
+// Configure the app instance.
+func init() {
+	app = cli.NewApp()
```

Comment:
Please use `utils.NewApp`, which sets version automatically. For its `gitCommit`, define a global variable called `gitCommit`. It will be filled in by the linker.

Please don't set `app.Copyright`.

Reply:
The reason why I avoided that is because `utils.NewApp` sets the version to the geth version. That might be misleading since geth versions might change while the functionality of this CLI will remain the same. (That's actually very likely given the simplicity of this tool.)

Reply:
All tools including rlpdump, evm, swarm use the same version. Releases in this repo are go-ethereum releases, not just geth releases.

Reply:
That's fair enough.

Reply:
Done.